src/cc/libbpf.c

```diff
@@ -70,13 +72,63 @@
 #define PERF_FLAG_FD_CLOEXEC (1UL << 3)
 #endif
 
+#define BCC_INSTANCE_FAIL -1
+#define BCC_INSTANCE_NOT_INIT -2
+#define BCC_INSTANCE_USE_DEFAULT -3
+static int bcc_instance_probe_cnt = BCC_INSTANCE_NOT_INIT;
+
 static int probe_perf_reader_page_cnt = 8;
 
 static uint64_t ptr_to_u64(void *ptr)
 {
   return (uint64_t) (unsigned long) ptr;
 }
 
+static int bcc_get_instance(char *buf, size_t buf_size) {
+  int res;
+  // An instance has already been created.
+  if (bcc_instance_probe_cnt >= 0) {
```

Comment:
Not really... `bcc_instance_probe_cnt` means either:
- The instance is just created, still in the process of attaching a probe. If the attaching succeeded the count would increase to 1.  
- The last probe from the instance was detached and the instance is in the process of being deleted.